### ISSUE TITLE
Switching accounts while uploading files doesn't update the main app file view after

### ISSUE THREAD

**Issue by strugee** [nextcloud/android#6227]
### Steps to reproduce
1. Have two accounts configured - let's call them A and B
2. Have A selected in the top-level file view
3. Share a picture with the Nextcloud app (i.e. upload with some other app's "share" button
4. In the upload folder chooser, switch to account B
5. Upload your picture

### Expected behaviour

I should see the files in account B in the main app view. Alternately, the account switcher accessible by an icon in the top-right corner should show account A as the active account.

### Actual behaviour

The account switcher in the top-right of the main app shows account B as the active account, but the actual files view shows data from account A.

### Can you reproduce this problem on https://try.nextcloud.com/?

N/A

### Environment data

Android version: 10

Device model: Pixel 3a XL

Stock or customized system: unlocked bootloader, but stock Android

Nextcloud app version: Nextcloud Dev 20200530 from F-Droid

Nextcloud server version: 18.0.4

Reverse proxy: Apache 2

#### Web server error log

N/A

#### Nextcloud log (data/nextcloud.log)

N/A

**Comment by tobiasKaminsky** [comment]
Upload is just a short account switching, this does not change which account is used during normal app usage.
(we change UI soon, so that you can see more easily which account is used)

**Comment by strugee** [comment]
@tobiasKaminsky hmm. That makes sense, but there's still a bug in dev edition 20200607. After switching accounts during uploading, the avatar shown in the top-right of the main file view is correct. However, when you tap on it, it shows the account you switched to during upload as the active account. (If that's hard to follow I can write a more detailed STR when I'm in front of a real computer 😅)

I'd like to reopen this.

**Comment by tobiasKaminsky** [comment]
> (If that's hard to follow I can write a more detailed STR when I'm in front of a real computer sweat_smile)

Yes, please :)

**Comment by AndyScherzinger** [comment]
still valid

**Comment by strugee** [comment]
Oof. Embarrassingly I forgot about this multiple times, despite reminders... but the STR in the description is finally updated to be clearer. So sorry for the wait :sweat_smile:

**Comment by strugee** [comment]
Removing the needs info label since the description has been updated, but feel free to apply it again and ping me if you need more information 👍
